src/d3a/d3a_core/util.py

```diff
@@ -388,14 +389,18 @@ def validate_const_settings_for_simulation():
         ConstSettings.IAASettings.MARKET_TYPE = 1
 
 
-def convert_datetime_to_str_keys(indict, outdict):
+def convert_datetime_to_str_keys(indict, outdict, ui_format=False):
     """
     Converts all Datetime keys in a dict into strings in DATE_TIME_FORMAT
     """
 
     for key, value in indict.items():
         if isinstance(key, DateTime):
-            outdict[key.format(DATE_TIME_FORMAT)] = indict[key]
+            if not ui_format:
+                outdict[key.format(DATE_TIME_FORMAT)] = indict[key]
+            else:
+                formatted_date = f"{calendar.month_name[key.month]} {key:%d %Y, %H:%M h}"
+                outdict[formatted_date] = indict[key]
```

Comment:
How about using pendulum instead:
`key.format("MMMM DD YYYY, HH:MM h")` ?

Reply:
You are right, I totally forgot about pendulum. Fixing this, thanks!